### PR TITLE
test(NODE-5898): small fixes and unskips of test TODOs

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

While auditing the existing skipped tests, I found a couple of quick fixes.  This PR unskips some tests that not longer fail, removes some unnecessary skip reason assignment, and removes some unnecessary TODO comments.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/integration/client-side-encryption/client_side_encryption.prose.test.js [hunk 1]
```diff
@@ -1174,7 +1174,6 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
 
         expect(insertError).to.be.instanceOf(MongoServerSelectionError);
 
-        // TODO(NODE-5296): check error.message once AggregateErrors are handled correctly
```
Comment (baileympearson) [inline test/integration/client-side-encryption/client_side_encryption.prose.test.js:1177]: This and the test below correctly check for `.message`.

File: test/integration/command-logging-and-monitoring/command_logging_and_monitoring.spec.test.ts [hunk 2]
```diff
@@ -3,22 +3,23 @@ import * as path from 'path';
...
+        // performed since it was only one document in the test. The test expectation
+        // is that we are using the bulk write result which was returned to the user
+        // as the reply in the command succeeded event instead of our raw reply from
+        // the server. There's nothing we can change here.
+        return description.includes(
+          'A successful unordered bulk write with an unacknowledged write concern'
+        )
+          ? `Test not applicable to Node.`
+          : false;
+      }
```
Comment (baileympearson) [inline test/integration/command-logging-and-monitoring/command_logging_and_monitoring.spec.test.ts:22]: just added a more applicable skip reason.

File: test/integration/crud/aggregation.test.ts [hunk 3]
```diff
@@ -332,7 +332,7 @@ describe('Aggregation', function () {
         });
       });
     }
-  ).skipReason = 'TODO(NODE-5617): aggregate explain tests failing on latest servers';
+  );
```
Comment (baileympearson) [inline test/integration/crud/aggregation.test.ts:335]: tests not failing.

File: test/integration/crud/explain.test.ts [hunk 4]
```diff
@@ -76,17 +75,6 @@ describe('CRUD API explain option', function () {
     collection = db.collection('test');
     await collection.insertOne({ a: 1 });
     commandStartedPromise = once(client, 'commandStarted');
-
```
Comment (baileympearson) [inline test/integration/crud/explain.test.ts:79]: tests not failing.

File: test/integration/load-balancers/load_balancers.spec.test.ts [hunk 5]
```diff
@@ -8,7 +8,6 @@ const filter = ({ description }) => {
     // Verified they use the same connection but the Node implementation executes
     // a getMore before the killCursors even though the stream is immediately
     // closed.
-    // TODO(NODE-3970): implement and reference a node specific integration test for this
```
Comment (baileympearson) [inline test/integration/load-balancers/load_balancers.spec.test.ts:11]: redundant comment

File: test/integration/node-specific/client_encryption.test.ts [hunk 6]
```diff
@@ -264,63 +264,6 @@ describe('ClientEncryption integration tests', function () {
       expect(rewrapManyDataKeyResult.bulkWriteResult).to.equal(undefined);
     });
 
-    it.skip(
```
Comment (baileympearson) [inline test/integration/node-specific/client_encryption.test.ts:267]: This isn't supported in our API so no reason to have this test.

File: test/integration/unified-test-format/unified_test_format.spec.test.ts [hunk 7]
```diff
@@ -3,12 +3,6 @@ import { runUnifiedSuite } from '../../tools/unified-spec-runner/runner';
 import { type TestFilter } from '../../tools/unified-spec-runner/schema';
 
 const filter: TestFilter = ({ description }) => {
-  if (description === 'unpin after transient error within a transaction and commit') {
-    // OLD COMMENT: commitTransaction retry seems to be swallowed by mongos in this case
-    // TODO(NODE-3943):
-    return `TODO(NODE-3943): commitTransaction retry seems to be swallowed by mongos in this case`;
-  }
-
   if (description === 'Client side error in command starting transaction') {
```
Comment (baileympearson) [inline test/integration/unified-test-format/unified_test_format.spec.test.ts:6]: both tests in this file are now passing.

File: test/tools/unified-spec-runner/runner.ts [hunk 8]
```diff
@@ -178,7 +178,6 @@ async function runUnifiedTest(
     if (topologyType === TopologyType.Sharded || topologyType === TopologyType.LoadBalanced) {
       for (const [, collection] of entities.mapOf('collection')) {
         try {
-          // TODO(NODE-4238): create / cleanup entities for each test suite
```
Comment (baileympearson) [inline test/tools/unified-spec-runner/runner.ts:181]: ticket is closed.

File: test/integration/command-logging-and-monitoring/command_logging_and_monitoring.spec.test.ts (hunk 2 above)
Comment (nbbeeken) [reply test/integration/command-logging-and-monitoring/command_logging_and_monitoring.spec.test.ts:22]: I do not fully understand why this doesn't apply to us. In a world where we were writing the driver from scratch could it obey this test? if so, could this _one day_, _allegedly_, be addressed?

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**